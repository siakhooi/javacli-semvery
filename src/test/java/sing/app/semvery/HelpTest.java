package sing.app.semvery;

import static com.github.stefanbirkner.systemlambda.SystemLambda.tapSystemOut;
import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;


@ExtendWith({SnapshotExtension.class})
public class HelpTest {
  private Expect expect;

  @Test
  void testPrintHelp() throws Exception {
    Parameters parameters = new Parameters();
    parameters.process(new String[] {"-h"});
    String text = tapSystemOut(() -> Help.printHelp(parameters));
    assertDoesNotThrow(() -> expect.toMatchSnapshot(text));
  }
}
