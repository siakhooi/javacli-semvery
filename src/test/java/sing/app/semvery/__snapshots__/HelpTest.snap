sing.app.semvery.HelpTest.testPrintHelp=[
semvery 0.5.0
Usage: semvery [options]
  Options:
    --operation, -o
      Options: [isValid, isStable]
    --version, -v
      Display version
    --help, -h
      Display help

Details:
  -o isValid version [version...]
  --operation isValid version [version...]

    Example:
      -o isValid 1.0.0
      -o isValid ABC
      -o isValid 1.0.0 2.0.0
      -o isValid 1.0.0 ABC

  -o isStable version [version...]
  --operation isStable version [version...]

    Example:
      -o isStable 1.0.0
      -o isStable 0.1.0
      -o isStable 1.0.0 0.1.0

]